srtcore/common.h

```diff
@@ -78,14 +78,19 @@ modified by
    #define NET_ERROR WSAGetLastError()
 #endif
 
-
 #ifdef _DEBUG
 #include <assert.h>
 #define SRT_ASSERT(cond) assert(cond)
 #else
 #define SRT_ASSERT(cond)
 #endif
 
+#if HAVE_FULL_CXX11
+#define SRT_STATIC_ASSERT(cond, msg) static_assert(cond, msg)
+#else
+#define SRT_STATIC_ASSERT(cond, msg)
```

Comment:
Consider add a simple implementation, for example:

```
template <bool>
class srt_static_assert;

template <>
class srt_static_assert<true> {
public:
    srt_static_assert(const char *) {}
};

#define SRT_STATIC_ASSERT(x, msg) srt_static_assert<(bool)(x)>(msg);
```

Reply:
Good suggestion, thank you!
There is one drawback though. This version can only be applied inside some function, not in a global space.

Example:
```c++
SRT_STATIC_ASSERT(SRTO_INPUTBW > SRTO_LATENCY, "input BW value"); // fails
int CUDTUnited::startup()
{
   SRT_STATIC_ASSERT(SRTO_INPUTBW > SRTO_LATENCY, "input BW value"); // ok
```


To be able to use it as a global space, e.g. the following modification may be done. But then there would be some unwanted variable.
```c++
#define SRT_STATIC_ASSERT(cond, msg) srt_static_assert<(bool)(cond)> xxx(msg);
```

In general, having only C++ `static_assert` is enough, such assertions are expected to be checked at least in GitHub Actions CI (with C++11 enabled).

Reply:
> In general, having only C++ static_assert is enough, such assertions are expected to be checked at least in GitHub Actions CI (with C++11 enabled).

Got it. By the way is there a roadmap to bump the required C++ version?

Reply:
I am afraid, no. There is a lot of projects still using C++03. :(